File: C/Q27RemoveElement.c
#include <stdio.h>

/*
Description:
Given an integer array nums and an integer val, remove all occurrences of val in nums in-place. 
The relative order of the elements may be changed.
Since it is impossible to change the length of the array in some languages, 
you must instead have the result be placed in the first part of the array nums. 
More formally, if there are k elements after removing the duplicates, 
then the first k elements of nums should hold the final result. 
It does not matter what you leave beyond the first k elements.
Return k after placing the final result in the first k slots of nums.
Do not allocate extra space for another array. 
You must do this by modifying the input array in-place with O(1) extra memory.

Example:
Input: nums = [3,2,2,3], val = 3
Output: 2, nums = [2,2,_,_]
Explanation: Your function should return k = 2, with the first two elements of nums being 2.
It does not matter what you leave beyond the returned k (hence they are underscores).
*/

int removeElement(int* nums, int numsSize, int val){
    int k = 0; 
    for(int i = 0; i < numsSize; i++){
        if(nums[i] != val){
            nums[k] = nums[i];
            k++;
        }
    }
    return k;
}

//Runtime: 10 ms, faster than 5.10% of C online submissions for Remove Element.
//Memory Usage: 5.9 MB, less than 96.03% of C online submissions for Remove Element.

File: Python/Q27RemoveElement.py
# @b-knd (jingru) on 08 August 2022 10:39:00

class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        k = 0;
        for i, value in enumerate(nums):
            if value != val:
                nums[k] = value
                k += 1
        return k

#Runtime: 17 ms, faster than 94.61% of Python online submissions for Remove Element.
#Memory Usage: 13.3 MB, less than 87.45% of Python online submissions for Remove Element.
